### PR TITLE
Add billing page link on billing modal

### PR DESCRIPTION
**Pull request by MauAraujo**
## What does this PR do?
Adds a link to the billing modal so users can visit the billing page, and not get blocked from using all of the dashboard. The modal now opens on the apps dashboard if the user doesn't have payment method or trial.

[apps-modal.webm](https://github.com/porter-dev/porter/assets/19579265/32397c14-6323-4a92-8d23-6f92d77a337f)

### CODE REVIEW

**Review by sunguroku — APPROVED** [review]

File: dashboard/src/main/home/app-dashboard/apps/Apps.tsx [hunk 1]
```diff
@@ -69,6 +77,24 @@ const Apps: React.FC = () => {
...
+    return timestampDate.isBefore(dayjs(new Date()));
+  };
+  const trialExpired =
+    plan !== null && plan && isTrialExpired(plan.trial_info.ending_before);
+
+  useEffect(() => {
+    if (trialExpired && !hasPaymentEnabled) {
+      setShowBillingModal(true);
+    }
+  });
```
Comment (ferozemohideen) [inline dashboard/src/main/home/app-dashboard/apps/Apps.tsx:96]: add [] as the second argument to useEffect

File: dashboard/src/main/home/app-dashboard/apps/Apps.tsx [hunk 2]
```diff
@@ -69,6 +77,24 @@ const Apps: React.FC = () => {
...
+
+  const isTrialExpired = (timestamp: string): boolean => {
+    if (timestamp === "") {
+      return true;
+    }
+    const timestampDate = dayjs(timestamp);
+    return timestampDate.isBefore(dayjs(new Date()));
+  };
+  const trialExpired =
+    plan !== null && plan && isTrialExpired(plan.trial_info.ending_before);
```
Comment (ferozemohideen) [inline dashboard/src/main/home/app-dashboard/apps/Apps.tsx:90]: plan != null && isTrialExpired()

**Review by ferozemohideen — APPROVED** [review]

File: dashboard/src/main/home/app-dashboard/apps/Apps.tsx (hunk 2 above)
Comment (MauAraujo) [reply dashboard/src/main/home/app-dashboard/apps/Apps.tsx]: thanks!

File: dashboard/src/main/home/app-dashboard/apps/Apps.tsx (hunk 1 above)
Comment (MauAraujo) [reply dashboard/src/main/home/app-dashboard/apps/Apps.tsx]: fixed!

**Pull request merged**